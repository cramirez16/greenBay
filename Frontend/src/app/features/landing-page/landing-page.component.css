.container {
  display: flex;
  flex-direction: column;
  text-align: center;
  gap: 20px;
  justify-content: space-evenly;
  height: calc(100vh - 64px - 30px);
  overflow: scroll;
  /* Hide scrollbar for IE, Edge and Firefox */
  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */
}

/* Hide scrollbar for Chrome, Safari and Opera */
.container::-webkit-scrollbar {
  display: none;
}

.header h1 {
  font-size: 24px;
  font-weight: bold;
  color: #3f51b5;
}

h2 {
  font-size: 18px;
  font-weight: bold;
  color: #3f51b5;
}

.header p {
  font-size: 16px;
  color: #3f51b5;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  font-size: 18px;
  color: #3f51b5;
  margin-bottom: 10px;
}

.description {
  color: #3f51b5;
  padding-left: 10%;
  padding-right: 10%;
}

.description p:first-child {
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 10px;
}

.description p:last-child {
  font-size: 14px;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 30px;
}

.card {
  padding: 20px;
  flex: 1;
  text-align: start;
}

mat-card {
  min-width: 300px;
  max-width: 400px;
  min-height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgb(242, 242, 242);
  color: #3f51b5;
}

@media (max-width: 900px) {
  .container {
    height: calc(100vh - 144px - 30px);
    padding-top: calc(144px + 30px);
  }
  .card-container {
    display: flex;
    flex-direction: column;
    align-content: center;
    flex-wrap: wrap;
    justify-content: center;
    gap: 30px;
  }
}
