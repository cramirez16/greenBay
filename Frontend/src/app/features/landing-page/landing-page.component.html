<section class="container">
    <!-- animate overlay effect when click -->
    <ul class="transition">
        <li></li>
        <li></li>
        <li></li>
        <li></li>
        <li></li>
    </ul>
    <div class="header">
        <h1>Welcome to Green Bay Service!</h1>
    </div>
    <h2>Our main features</h2>
    <mat-divider></mat-divider>
    <div class="card-container">
        <mat-card class="card">
            <mat-card-title>Registration</mat-card-title>
            <mat-card-content>
                <p>
                    Sign up and create an account to access the Green Bay application.
                    Provide your details and get ready to explore!
                </p>
            </mat-card-content>
        </mat-card>
        <mat-card class="card">
            <mat-card-title>Login</mat-card-title>
            <mat-card-content>
                <p>
                    Already have an account? Log in to access your personalized dashboard,
                    manage your items and bids. Enjoy a seamless experience.
                </p>
            </mat-card-content>
        </mat-card>
        <mat-card class="card">
            <mat-card-title>Purchase</mat-card-title>
            <mat-card-content>
                <p>
                    Search, find and purchase amazing oportunities.
                </p>
            </mat-card-content>
        </mat-card>
    </div>
    <mat-divider></mat-divider>
    <div class="description">
        <p>Description:</p>
        <p>
            Our Green Bay application is designed to provide you with a convenient
            platform to explore and find a wide variety for a wide range of events. Whether
            you're looking for concerts, sports events, theater performances, or other
            entertainment options, our application has got you covered!
        </p>
    </div>
</section>