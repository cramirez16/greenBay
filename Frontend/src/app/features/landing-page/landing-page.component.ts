import { Component, OnInit } from '@angular/core';
import { gsap } from 'gsap';

@Component({
  selector: 'app-landing-page',
  templateUrl: './landing-page.component.html',
  styleUrls: ['./landing-page.component.css'],
})
export class LandingPageComponent implements OnInit {
  name = 'Angular';

  ngOnInit() {
    this.setupGsap();
  }

  setupGsap(): void {
    this.pageTransition();
    // this.contentAnimation();
  }

  pageTransition() {
    let tl = gsap.timeline();
    tl.to('ul.transition li', {
      duration: 2,
      scaleY: 0,
      transformOrigin: 'bottom left',
      stagger: 0.1,
      delay: 0.1,
    });
  }
}
