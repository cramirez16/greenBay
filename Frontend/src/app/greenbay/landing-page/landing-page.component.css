.container {
  text-align: center;
  padding: 50px 30px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  justify-content: top;
  height: 100vh;
  overflow: scroll;
}

.header h1 {
  font-size: 24px;
  font-weight: bold;
  color: #3f51b5;
}

h2 {
  font-size: 18px;
  font-weight: bold;
  color: #3f51b5;
}

.header p {
  font-size: 16px;
  color: #3f51b5;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  font-size: 18px;
  color: #3f51b5;
  margin-bottom: 10px;
}

.description {
  color: #3f51b5;
}

.description p:first-child {
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 10px;
}

.description p:last-child {
  font-size: 14px;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 30px;
}

.card {
  padding: 20px;
  flex: 1;
  text-align: start;
}

mat-card {
  min-width: 300px;
  max-width: 400px;
  min-height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
}

@media (max-width: 900px) {
  .card-container {
    display: flex;
    flex-direction: column;
    align-content: center;
    flex-wrap: wrap;
    justify-content: center;
    gap: 30px;
  }
}
