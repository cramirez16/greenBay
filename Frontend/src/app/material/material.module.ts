import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatCardModule } from '@angular/material/card';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatDialogModule } from '@angular/material/dialog';
import { MatDividerModule } from '@angular/material/divider';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatGridListModule } from '@angular/material/grid-list';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatMenuModule } from '@angular/material/menu';
import { MatNativeDateModule } from '@angular/material/core';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSelectModule } from '@angular/material/select';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatTableModule } from '@angular/material/table';
import { MatToolbarModule } from '@angular/material/toolbar';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgIf, NgFor } from '@angular/common';

@NgModule({
  imports: [NgIf, NgFor],
  exports: [
    MatButtonModule,
    MatCardModule,
    MatCheckboxModule,
    MatDatepickerModule,
    MatDialogModule,
    MatDividerModule,
    MatFormFieldModule,
    MatGridListModule,
    MatIconModule,
    MatInputModule,
    MatMenuModule,
    MatPaginatorModule,
    MatNativeDateModule,
    MatSelectModule,
    MatSnackBarModule,
    MatTableModule,
    MatToolbarModule,
    FormsModule,
    NgIf,
    NgFor,
    ReactiveFormsModule,
  ],
})
export class MaterialModule {}
