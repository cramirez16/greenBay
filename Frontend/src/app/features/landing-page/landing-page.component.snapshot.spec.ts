import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LandingPageComponent } from './landing-page.component';
import { MaterialModule } from '../../material/material.module';

describe('LandingPageComponent', () => {
  let component: LandingPageComponent;
  let fixture: ComponentFixture<LandingPageComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LandingPageComponent],
      imports: [MaterialModule],
    });
    fixture = TestBed.createComponent(LandingPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('[snapshot] should renders markup to snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
