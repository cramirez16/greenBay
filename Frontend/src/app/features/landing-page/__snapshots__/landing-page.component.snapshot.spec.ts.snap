// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageComponent [snapshot] should renders markup to snapshot 1`] = `
<app-landing-page>
  <section
    class="container"
  >
    <div
      class="header"
    >
      <h1>
        Welcome to Green Bay Service!
      </h1>
    </div>
    <h2>
      Our main features
    </h2>
    <mat-divider
      aria-orientation="horizontal"
      class="mat-divider mat-divider-horizontal"
      role="separator"
    />
    <div
      class="card-container"
    >
      <mat-card
        class="mat-mdc-card mdc-card card"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          Registration
        </mat-card-title>
        <mat-card-content
          class="mat-mdc-card-content"
        >
          <p>
             Sign up and create an account to access the Green Bay application. Provide your details and get ready to explore! 
          </p>
        </mat-card-content>
      </mat-card>
      <mat-card
        class="mat-mdc-card mdc-card card"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          Login
        </mat-card-title>
        <mat-card-content
          class="mat-mdc-card-content"
        >
          <p>
             Already have an account? Log in to access your personalized dashboard, manage your items and bids. Enjoy a seamless experience. 
          </p>
        </mat-card-content>
      </mat-card>
      <mat-card
        class="mat-mdc-card mdc-card card"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          Purchase
        </mat-card-title>
        <mat-card-content
          class="mat-mdc-card-content"
        >
          <p>
             Search, find and purchase amazing oportunities. 
          </p>
        </mat-card-content>
      </mat-card>
    </div>
    <mat-divider
      aria-orientation="horizontal"
      class="mat-divider mat-divider-horizontal"
      role="separator"
    />
    <div
      class="description"
    >
      <p>
        Description:
      </p>
      <p>
         Our Green Bay application is designed to provide you with a convenient platform to explore and find a wide variety for a wide range of events. Whether you're looking for concerts, sports events, theater performances, or other entertainment options, our application has got you covered! 
      </p>
    </div>
  </section>
</app-landing-page>
`;
